java/code/src/com/redhat/rhn/manager/system/SystemManager.java

```diff
@@ -1227,7 +1227,12 @@ public static DataResult<VirtualSystemOverview> virtualSystemsInSet(
      * @return list of SystemGroupOverviews.
      */
     public static DataResult<SystemGroupOverview> groupList(User user, PageControl pc) {
-        SelectMode m = ModeFactory.getMode("SystemGroup_queries", "visible_to_user");
+        String query = "visible_to_user";
```

Comment:
I think we may not need a new query and can play around with the elaboratores in the SelectMode query instead. this way we would avoid duplicating the code.
Calling `m.getElaborators()` will return the elaborator, and then we can manipulate the list.